build_tools/install_dependencies.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+
+set -e
+
+J=-j8
+
+## Install static dependencies for compiling static rocksdbjava with cmake.
+## Note that .a are made with -fPIC, as they end up in a .so
+## Also note that this should be enhanced to arc specific defines.
+## generate cmake flags
+
+CMAKEFLAGS="-DWITH_JNI=OFF -DWITH_LIBURING=OFF -DWITH_LZ4=OFF -DWITH_SNAPPY=OFF -DWITH_ZLIB=OFF -DWITH_ZSTD=OFF -DWITH_BZ2=OFF -DWITH_GFLAGS=OFF -DWITH_JEMALLOC=OFF"
+
+## For mac
+case "$ARCH" in
+    x86_64 ) ARCH_CFLAGS="-arch x86_64";;
+    arm64 ) ARCH_CFLAGS="-arch arm64";;
+esac
+
+
+usage ()
+{
+    echo ' installs dependencies, and generate cmake flags for rocksdb compilation.
+    ./build_tools/install_deps.sh <dirname> packages ...
+    cmake $(cat <dirname>/cmake-flags) ...
+    packages:'
+    sed -ne 's/^with_\(.*\)().*#\(.*\)/\t\1:    \t\2/p' < $0
+}
+
+cmakeflags()
+{
+    NAME=WITH_$1
+    shift
+    CMAKEFLAGS=${CMAKEFLAGS//-D$NAME=OFF/}
+    CMAKEFLAGS="$CMAKEFLAGS -D$NAME=ON $@"
+}
+
+download ()
+{
+    file=$1
+    url=$2
+    if [ -f "$file" ]
+    then
+        echo "Reusing downloaded file $file, delete if it's corrupt." >&2
+    else
+        curl --fail --silent --show-error --location --output $file $url
+    fi
+}
+
+with_iouring()  # makes liburing.a with -fPIC
+{
+    echo Compiling iouring to $DIR/iouring/ >&2
+    cd $DIR/
+    download liburing-0.7.tar.gz https://github.com/axboe/liburing/archive/refs/tags/liburing-0.7.tar.gz
+    rm -fr liburing-liburing-0.7 $DIR/iouring/
+    tar xf liburing-0.7.tar.gz
+    cd liburing-liburing-0.7/
+    ./configure --prefix=$DIR/iouring/
+    make $J V=1 -C src ENABLE_SHARED=0 CFLAGS=" -g -fomit-frame-pointer -O2 -fPIC" install
```

Comment:
Why `-g`?

Reply:
`-g` enables debug info, and is orthogonal to `-O` ... it's just extra bytes in the obj file, which may be useful in case of stack dump